src/main/java/de/tum/in/www1/artemis/service/WebsocketMessagingService.java

```diff
@@ -80,20 +80,20 @@ public void broadcastNewResult(Participation participation, Result result) {
         // TODO: Are there other cases that must be handled here?
         if (participation instanceof StudentParticipation studentParticipation) {
             final Exercise exercise = studentParticipation.getExercise();
-            final boolean isWorkingPeriodOver;
+            boolean isWorkingPeriodOver;
             if (exercise.isExamExercise()) {
-                isWorkingPeriodOver = examDateService.isExerciseWorkingPeriodOver(exercise);
+                isWorkingPeriodOver = examDateService.isExerciseWorkingPeriodOver(exercise, studentParticipation);
             }
             else {
                 isWorkingPeriodOver = exerciseDateService.isAfterLatestDueDate(exercise);
             }
             // Don't send students results after the exam ended
-            boolean isAfterExamEnd = isWorkingPeriodOver && exercise.isExamExercise();
+            boolean isAfterExamEnd = isWorkingPeriodOver && exercise.isExamExercise() && !exercise.getExamViaExerciseGroupOrCourseMember().isTestExam();
             // If the assessment due date is not over yet, do not send manual feedback to students!
-            boolean isReadyForRelease = AssessmentType.AUTOMATIC == result.getAssessmentType() || exercise.getAssessmentDueDate() == null
+            boolean isAutomaticAssessmentOrDueDateNotOverYet = AssessmentType.AUTOMATIC == result.getAssessmentType() || exercise.getAssessmentDueDate() == null
                     || ZonedDateTime.now().isAfter(exercise.getAssessmentDueDate());
```

Comment:
No, we only want to broadcast the new result after the assessment due date (this is in case of a live assessment). Before the assessment due date, the result remains hidden.

Reply:
Hm. It was before like this though. Probably we just added the wrong name to the variable ..?

Reply:
The current version makes sense. We want to distribute the result only if the result is automatic, or, if it is is manual, if the assessment due date has passed. Therefore we check if the current time is after the assessment due date, and only then we enter the if block below

Reply:
I only went by the variable name and got confused. The underlying logic should be right after thinking a bit